### PR TITLE
[epforwarder] Update NetFlow default configs

### PR DESCRIPTION
**Pull request by AlexandreYang**
Depend on https://github.com/DataDog/datadog-agent/pull/12933

### What does this PR do? + Motivations

[epforwarder] Update NetFlow default configs

1/ Revert `defaultBatchMaxContentSize` to 5Mo (20Mo requires backend intake changes, currently if payload >5Mo is send, intake will return "content too large" error)

2/ Update defaultBatchMaxSize to 10k, since NetFlow traffic is about high volume of small messages. Higher BatchMaxSize will reduce the number of payloads sent to intake.

3/ Update defaultInputChanSize to 10k
```
// High input chan is needed to handle high number of flows being flushed by NetFlow Server every 10s
// Customers might need to set `network_devices.forwarder.input_chan_size` to higher value if flows are dropped
// due to input channel being full.
// TODO: A possible better solution is to make SendEventPlatformEvent blocking when input chan is full and avoid
//   dropping events. This can't be done right now due to SendEventPlatformEvent being called by
//   aggregator loop, making SendEventPlatformEvent blocking might slow down other type of data handled
//   by aggregator.
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/update_netflow_default_configs-fe0c68eb55dcf012.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    [epforwarder] Update NetFlow default configs (1. Revert defaultBatchMaxContentSize to 5Mo,
```
Comment (maycmlee) [inline releasenotes/notes/update_netflow_default_configs-fe0c68eb55dcf012.yaml:11]:
```suggestion
    [epforwarder] Update NetFlow default configs to: 1. Revert `defaultBatchMaxContentSize` to 5Mo,
```

File: releasenotes/notes/update_netflow_default_configs-fe0c68eb55dcf012.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    2. Update defaultBatchMaxSize to 10k, 3. Update defaultInputChanSize to 100k)
```
Comment (maycmlee) [inline releasenotes/notes/update_netflow_default_configs-fe0c68eb55dcf012.yaml:12]:
```suggestion
    2. Update `defaultBatchMaxSize` to 10k, 3. Update `defaultInputChanSize` to 100k.
```

**Review by maycmlee — COMMENTED** [review]
Left some suggestions!

**Review by djmitche — APPROVED** [review]

**Review by maycmlee — APPROVED** [review]

File: pkg/epforwarder/epforwarder.go [hunk 3]
```diff
@@ -215,7 +226,7 @@ func newHTTPPassthroughPipeline(desc passthroughPipelineDesc, destinationsContex
 	if endpoints.BatchMaxSize <= pkgconfig.DefaultBatchMaxSize {
 		endpoints.BatchMaxSize = desc.defaultBatchMaxSize
 	}
-	if endpoints.InputChanSize <= 0 {
+	if endpoints.InputChanSize <= pkgconfig.DefaultInputChanSize {
```
Comment (AlexandreYang) [inline pkg/epforwarder/epforwarder.go:229]: Note: Need to use similar logic than `BatchMaxSize` and `BatchMaxContentSize` above. 
At this point `endpoints.InputChanSize` contain the default value set in config.go, comparing `<= 0` won't lead to expected result that is using `pkgconfig.DefaultInputChanSize` by default.

**Comment by AlexandreYang** [comment]
@djmitche I made few changes since your last review, do you mind reviewing the PR again ?

- Added this fix https://github.com/DataDog/datadog-agent/pull/12934/commits/6af9fc4853aeb187a6786075e50917a6a3685ff4
- Updated comment and default value for `defaultInputChanSize` https://github.com/DataDog/datadog-agent/pull/12934/files#diff-6569a3f8ce509bc4602e1db6ae7933bbf5079532cbad740507d6dfe7a6fba3f4R107-R114

File: pkg/epforwarder/epforwarder.go [hunk 4]
```diff
@@ -98,9 +98,20 @@ var passthroughPipelineDescs = []passthroughPipelineDesc{
 		hostnameEndpointPrefix:        "ndmflow-intake.",
 		intakeTrackType:               "ndmflow",
 		defaultBatchMaxConcurrentSend: 10,
-		defaultBatchMaxContentSize:    20e6, // max 20Mb uncompressed size per payload
-		defaultBatchMaxSize:           pkgconfig.DefaultBatchMaxSize,
-		defaultInputChanSize:          pkgconfig.DefaultInputChanSize,
+		defaultBatchMaxContentSize:    pkgconfig.DefaultBatchMaxContentSize,
```
Comment (ken-schneider) [inline pkg/epforwarder/epforwarder.go:101]: What's the default coming from `pkgconfig` here?

**Review by ken-schneider — APPROVED** [review]

File: pkg/epforwarder/epforwarder.go (hunk 4 above)
Comment (AlexandreYang) [reply pkg/epforwarder/epforwarder.go:101]: It's 5Mb.

**Pull request merged**